src/aosm/azext_aosm/cli_handlers/onboarding_base_handler.py

```diff
@@ -42,14 +43,21 @@ def output_folder_file_name(self) -> str:
     def __init__(self, input_json: str | None = None):
         # Config may be optional (to generate blank config file)
         input_json_path = Path(input_json) if input_json else None
-        config_dict = (
-            self._read_config_from_file(input_json_path) if input_json_path else {}
-        )
+        print(input_json_path)
         # Ensure config is of correct type
         try:
-            self.config = self._get_config(config_dict)
+            if input_json_path.suffix == '.jsonc':
+                config_dict = (
+                    self._read_config_from_file(input_json_path) if input_json_path else {}

```

Comment:
The ordering of these `if` statements doesn't make sense to me 😕
First we check the suffix is ".jsonc". Only if that's true, we then check if it exists at all (`if input_json_path else`) - does this need another look?